apps/react/src/features/user/components/AvatarUpload/AvatarUpload.tsx

```diff
@@ -0,0 +1,119 @@
+import { FC, useRef, useState } from 'react';
+import { Alert, Avatar, Box, Button } from '@mui/material';
+
+import { typedMemo } from '@js-camp/react/utils/typedMemo';
+
+import styles from './AvatarUpload.module.css';
+
+interface Props {
+
+	/** Avatar image url. */
+	readonly imageUrl: string | null;
+
+	/** Change avatar image. */
+	readonly changeImage: (image: File | null) => void;
+
+	/** Is disable. */
+	readonly disable?: boolean;
+
+	/** Class name. */
+	readonly className?: string;
+}
+
+/** Max image file size - 60 Kb. */
+const MAX_IMAGE_FILE_SIZE = 1024 * 60;
+
+/** Accept image file types. */
+const ACCEPT_FILE_TYPES = ['image/jpeg', 'image/png', 'image/webp'];
+
+/** Error display time. */
+const ERROR_DISPLAY_TIME = 1000 * 3;
+
+/**
+ * Avatar upload component.
+ * @param props Props.
+ */
+const AvatarUploadComponent: FC<Props> = props => {
+	const [error, setError] = useState<string | null>(null);
+	const errorTimer = useRef<NodeJS.Timeout | null>(null);
+
+	/**
+	 * Upload avatar image.
+	 * @param event Change event.
+	 */
+	const uploadImage = (event: React.ChangeEvent<HTMLInputElement>): void => {
+		clearImageError();
+
+		const file = event.target.files?.[0];
+		if (file === undefined) {
+			return;
+		}
+
+		const imageError = checkImageErrors(file);
+		if (imageError !== null) {
+			setImageErrorWithTimer(imageError);
+			return;
+		}
+
+		props.changeImage(file);
+	};
+
+	/** Clear image error. */
+	const clearImageError = (): void => {
+		setError(null);
+		errorTimer.current = null;
+	};
+
+	/**
+	 * Check image errors.
+	 * @param file Uploaded file.
+	 */
+	const checkImageErrors = (file: File): string | null => {
+		if (!ACCEPT_FILE_TYPES.includes(file.type)) {
+			return 'Accept image type: jpeg/jpg, png, webp';
+		}
+		if (file.size > MAX_IMAGE_FILE_SIZE) {
+			return 'Max file size - 60 Kb';
+		}
+		return null;
+	};
+
+	/**
+	 * Set image error with timer.
+	 * @param imageError Image error.
+	 */
+	const setImageErrorWithTimer = (imageError: string): void => {
+		setError(imageError);
+		errorTimer.current = setTimeout(clearImageError, ERROR_DISPLAY_TIME);
+	};
+
+	/** Delete image. */
+	const deleteImage = (): void => {
+		props.changeImage(null);
+	};
+
+	return (
+		<Box className={`${styles['avatar-upload']} ${props.className ?? ''}`}>
+			<Avatar alt="Current user avatar"
+				className={styles['avatar-upload__avatar']}
+				src={props.imageUrl ?? undefined}/>
+			{!props.disable &&
+				<Box className={styles['avatar-upload__actions']}>
+					<Button>
+						Upload
+						<input type="file"
+							onChange={uploadImage}
+							accept={ACCEPT_FILE_TYPES.join(', ')}
+							className={styles['avatar-upload__input']}/>
+					</Button>
```

Comment:
I guess using `button` here is not semantically correct, but you can easily fix by doing this:
```suggestion
					<Button component="div">
						Upload
						<input type="file"
							onChange={uploadImage}
							accept={ACCEPT_FILE_TYPES.join(', ')}
							className={styles['avatar-upload__input']}/>
					</Button>
```
or
```suggestion
					<Button component={Box}>
						Upload
						<input type="file"
							onChange={uploadImage}
							accept={ACCEPT_FILE_TYPES.join(', ')}
							className={styles['avatar-upload__input']}/>
					</Button>
```